Zad_3_4_5.sc

```diff
@@ -0,0 +1,100 @@
+
+
+trait Debug {
+  //Zad 3 (5 pkt)
+  def debugName(): Unit = {
+    val class_name = this.getClass.getName
+    val name = class_name.split("\\$").last
+    println("Class: " + name)
+  }
+
+  //Zad 4 (5pkt)
+  def debugVars(): Unit = {
+    val fields = this.getClass.getDeclaredFields
+    for (field <- fields) {
+      field.setAccessible(true)
+      val str = "Var: " + field.getName + " => " + field.getType + ", " + field.get(this)
```

Comment:
str to mało informatywna nazwa dla zmiennej. lepiej nazwać ją na styl fieldDetails